### PR TITLE
Code refine.

### PR DESCRIPTION
**Pull request by Roy-AMD**
Encapsulate repeated code into functions to reduce code length.
Use a more reliable way to obtain the access name.

### CODE REVIEW

**Comment by johnkslang** [comment]
Good reduction of replicated code.

File: glslang/MachineIndependent/iomapper.cpp [hunk 1]
```diff
@@ -907,6 +892,17 @@ void TDefaultGlslIoResolver::reserverResourceSlot(TVarEntryInfo& ent, TInfoSink&
     }
 }
 
+TString& TDefaultGlslIoResolver::getAccessName(const TIntermSymbol* symbol)
+{
+    TString name;
+    if (symbol->getBasicType() == EbtBlock) {
```
Comment (johnkslang) [inline glslang/MachineIndependent/iomapper.cpp:898]: The test is changing from `IsAnonymous` to whether the basic type is a block.  Are these equivalent?

Comment (Roy-AMD) [reply glslang/MachineIndependent/iomapper.cpp:898]: > The test is changing from `IsAnonymous` to whether the basic type is a block. Are these equivalent?

In a sense, it is not equivalent. When a block is named, the new code will still get the type name, but this is exactly what we expected. GlslIoMapper has not completely stabilized. We are still looking for bugs. When we get a stable version and we will provide test code for it. But now these codes may still be changed more frequently.

Comment (Andreyogld3d) [reply glslang/MachineIndependent/iomapper.cpp:898]: @Roy-AMD , @johnkslang I have warning during compilation:

Visual Studio 2017:
glslang\glslang\machineindependent\iomapper.cpp(903): warning C4172: returning address of local variable or temporary: name

Comment (johnkslang) [reply glslang/MachineIndependent/iomapper.cpp:898]: Oh, right. This is not good code.

Probably, it should directly return the references it gets, or declare name to be a reference.  I'll look.

Comment (johnkslang) [reply glslang/MachineIndependent/iomapper.cpp:898]: I made it simply be;

```
    return symbol->getBasicType() == EbtBlock ?
        symbol->getType().getTypeName() :
        symbol->getName();
```

And also fixed `const` correctness.

**Pull request merged**